docs/0.5/howto/configure_logging.md

```diff
@@ -0,0 +1,230 @@
+wheat from the chaff. This document contains a high level description of how
+logging works in the context of Splinter, examples of logging configurations,
+and trouble shooting tips for common use cases.
+
+## Prerequisites
+
+* A working Splinter installation.
+* Edit access to the `splinterd.toml` file for the above Splinter installation.
+
+## High Level Logging Overview
+
+There are two main abstractions built into our logging solution.
+
+* Loggers
+* Appenders
+
+Loggers are the interface between Splinter code and appenders. Appenders are
+the interface between Loggers and where logs are actually written.
+
+There is a many to many relationship between the two concepts, each logger can
+relate to any number of appenders and each appender can relate to any number of
+loggers.
+
+The life cycle of log flows as follows. Log statement is called in Splinter, the
+message and associated metadata is directed to a logger. The logger forwards the
+log to all of its configured appenders. Each of those appenders write the log to
+their resource.
+
+Besides their forwarding and writing duties loggers and appenders also filter
+the logs they receive by level.
+
+#### Log Levels
+
+* Error
+* Warn
+* Info,
+* Debug
+* Trace
+
+Listed above in decreasing order of importance and volume. Each level includes
+all the higher levels i.e. `Info` includes messages with levels `Warn`, `Error`,
+and `Info`. 
+
+## Loggers in Depth
+
+Loggers form a tree structure. The root logger is the root of the logger tree
+with all other loggers being its descendants. As such any log message that is
+not directed to any other logger will be handled by the root logger.
+
+Log messages are directed to loggers based on the loggers name and where the log
+message originated. So log messages from the `splinter` package will match to
+any loggers named `splinter`. Log messages match to the most specific
+package/logger available so a message from `splinter::config` would match to a
+logger named `splinter::config` if its available before matching to a `splinter`
+logger.
+
+Loggers have two other fields `appenders` and `level`. Appenders is an array of
+appender names and serves as the connection between loggers and appenders. The
+`level` field controls the filtering functionality of the logger as discussed
+above.
+
+For non root loggers the `appenders` and `level` fields are optional. The level
+will be set to the root loggers level. Appenders are a little more complicated,
+each logger inherits its parents appenders. So if you had two loggers, `root`
+and `splinter` and `root` has both `stdout` and `stderr` appenders configured.
+Then the `splinter` logger would also have the same `stdout` and `stderr`
+appenders plus any specifically called out in its configuration.
+
+## Appenders in Depth
+
+There are several types of appenders denoted by the `kind` field.
+
+#### Appender Kinds
+* stdout
+* stderr
+* file
```

Comment:
If file is available, it needs an example in the example config as well.

Reply:
There was already an example with file but there wasn't one with rolling_file so I added one. I think that is what you meant?

Reply:
No, I mean in the example config file, not this PR.